### ISSUE TITLE
YError in 3.1.0

### ISSUE THREAD

**Issue by deisi** [vega/altair#1606]
Hey, its me again with the YErrors. I know #1473 was fixed in #1480.
However I just wanted to use 3.1.0 with:

``` python
df = pd.DataFrame(
    [[0, 1,3], [1, 1, 2], [2, 3, 1]],
    columns=['index','data', 'error']
)

base = alt.Chart(df).encode(
    alt.X('index'),
    alt.Y('data'),
    alt.YError('error'),
    #YError='error'
)
chart = base.mark_errorbar(clip=True) + base.mark_point(size=200, filled=True,)
chart = chart.configure_axis(labelFontSize=20).configure_legend(labelFontSize=20)
chart
```

But the final chart call gives:
```
SchemaValidationError                     Traceback (most recent call last)
~/.local/lib/python3.7/site-packages/altair/vegalite/v3/api.py in to_dict(self, *args, **kwargs)
    376         if dct is None:
    377             kwargs['validate'] = 'deep'
--> 378             dct = super(TopLevelMixin, copy).to_dict(*args, **kwargs)
    379 
    380         # TODO: following entries are added after validation. Should they be validated?

~/.local/lib/python3.7/site-packages/altair/utils/schemapi.py in to_dict(self, validate, ignore, context)
    283             result = _todict({k: v for k, v in self._kwds.items()
    284                               if k not in ignore},
--> 285                               validate=sub_validate, context=context)
    286         else:
    287             raise ValueError("{} instance has both a value and properties : "

~/.local/lib/python3.7/site-packages/altair/utils/schemapi.py in _todict(obj, validate, context)
     48         return [_todict(v, validate, context) for v in obj]
     49     elif isinstance(obj, dict):
---> 50         return {k: _todict(v, validate, context) for k, v in obj.items()
     51                 if v is not Undefined}
     52     elif hasattr(obj, 'to_dict'):

~/.local/lib/python3.7/site-packages/altair/utils/schemapi.py in <dictcomp>(.0)
     49     elif isinstance(obj, dict):
     50         return {k: _todict(v, validate, context) for k, v in obj.items()
---> 51                 if v is not Undefined}
     52     elif hasattr(obj, 'to_dict'):
     53         return obj.to_dict()

~/.local/lib/python3.7/site-packages/altair/utils/schemapi.py in _todict(obj, validate, context)
     46         return obj.to_dict(validate=validate, context=context)
     47     elif isinstance(obj, (list, tuple)):
---> 48         return [_todict(v, validate, context) for v in obj]
     49     elif isinstance(obj, dict):
     50         return {k: _todict(v, validate, context) for k, v in obj.items()

~/.local/lib/python3.7/site-packages/altair/utils/schemapi.py in <listcomp>(.0)
     46         return obj.to_dict(validate=validate, context=context)
     47     elif isinstance(obj, (list, tuple)):
---> 48         return [_todict(v, validate, context) for v in obj]
     49     elif isinstance(obj, dict):
     50         return {k: _todict(v, validate, context) for k, v in obj.items()

~/.local/lib/python3.7/site-packages/altair/utils/schemapi.py in _todict(obj, validate, context)
     44     """Convert an object to a dict representation."""
     45     if isinstance(obj, SchemaBase):
---> 46         return obj.to_dict(validate=validate, context=context)
     47     elif isinstance(obj, (list, tuple)):
     48         return [_todict(v, validate, context) for v in obj]

~/.local/lib/python3.7/site-packages/altair/vegalite/v3/api.py in to_dict(self, *args, **kwargs)
    376         if dct is None:
    377             kwargs['validate'] = 'deep'
--> 378             dct = super(TopLevelMixin, copy).to_dict(*args, **kwargs)
    379 
    380         # TODO: following entries are added after validation. Should they be validated?

~/.local/lib/python3.7/site-packages/altair/utils/schemapi.py in to_dict(self, validate, ignore, context)
    283             result = _todict({k: v for k, v in self._kwds.items()
    284                               if k not in ignore},
--> 285                               validate=sub_validate, context=context)
    286         else:
    287             raise ValueError("{} instance has both a value and properties : "

~/.local/lib/python3.7/site-packages/altair/utils/schemapi.py in _todict(obj, validate, context)
     48         return [_todict(v, validate, context) for v in obj]
     49     elif isinstance(obj, dict):
---> 50         return {k: _todict(v, validate, context) for k, v in obj.items()
     51                 if v is not Undefined}
     52     elif hasattr(obj, 'to_dict'):

~/.local/lib/python3.7/site-packages/altair/utils/schemapi.py in <dictcomp>(.0)
     49     elif isinstance(obj, dict):
     50         return {k: _todict(v, validate, context) for k, v in obj.items()
---> 51                 if v is not Undefined}
     52     elif hasattr(obj, 'to_dict'):
     53         return obj.to_dict()

~/.local/lib/python3.7/site-packages/altair/utils/schemapi.py in _todict(obj, validate, context)
     44     """Convert an object to a dict representation."""
     45     if isinstance(obj, SchemaBase):
---> 46         return obj.to_dict(validate=validate, context=context)
     47     elif isinstance(obj, (list, tuple)):
     48         return [_todict(v, validate, context) for v in obj]

~/.local/lib/python3.7/site-packages/altair/utils/schemapi.py in to_dict(self, validate, ignore, context)
    291                 self.validate(result)
    292             except jsonschema.ValidationError as err:
--> 293                 raise SchemaValidationError(self, err)
    294         return result
    295 

SchemaValidationError: Invalid specification

        altair.vegalite.v3.schema.core.FacetedEncoding, validating 'additionalProperties'

        Additional properties are not allowed ('yerror' was unexpected)
        

LayerChart({
  config: Config({
    axis: AxisConfig({
      labelFontSize: 20
    }),
    legend: LegendConfig({
      labelFontSize: 20
    })
  }),
  data:    index  data  error
  0      0     1      3
  1      1     1      2
  2      2     3      1,
  layer: [Chart({
    encoding: FacetedEncoding({
      x: X({
        field: 'index',
        type: 'quantitative'
      }),
      y: Y({
        field: 'data',
        type: 'quantitative'
      }),
      yerror: YError({
        field: 'error'
      })
    }),
    mark: ErrorBarDef({
      clip: True,
      type: 'errorbar'
    })
  }), Chart({
    encoding: FacetedEncoding({
      x: X({
        field: 'index',
        type: 'quantitative'
      }),
      y: Y({
        field: 'data',
        type: 'quantitative'
      }),
      yerror: YError({
        field: 'error'
      })
    }),
    mark: MarkDef({
      filled: True,
      size: 200,
      type: 'point'
    })
  })]
})
```

**Comment by jakevdp** [comment]
This error is due to #1573, which was fixed in #1597. That fix has not yet made it into a release. The workaround for the moment is to fully-specify the encoding, e.g.
```
encode(
    yError=alt.YError('column:Q')
)
```

**Comment by deisi** [comment]
aha this is it. THX
